src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocSummaryLocationCheckTest.java

```diff
@@ -0,0 +1,78 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2019 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.checks.javadoc;
+
+import static com.puppycrawl.tools.checkstyle.checks.javadoc.JavadocSummaryLocationCheck.MSG_JAVADOC_SUMMARY_FIRST_LINE;
+import static com.puppycrawl.tools.checkstyle.checks.javadoc.JavadocSummaryLocationCheck.MSG_JAVADOC_SUMMARY_SECOND_LINE;
+import static org.junit.Assert.assertArrayEquals;
+
+import org.junit.Test;
+
+import com.puppycrawl.tools.checkstyle.AbstractModuleTestSupport;
+import com.puppycrawl.tools.checkstyle.DefaultConfiguration;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+
+public class JavadocSummaryLocationCheckTest extends AbstractModuleTestSupport {
```

Comment:
I don't see any examples like where first 2 lines are empty.
````
/**
*
* Example.
 */
````
Should this be a violation for second since it is on the 3rd line?

Reply:
done. And some more examples: with missing asterisks, with blank lines only, without the summary.

Reply:
Yes, it should. Added to test cased.